crates/diesel_models/src/user.rs

```diff
@@ -58,6 +67,11 @@ pub enum UserUpdate {
         is_verified: Option<bool>,
         preferred_merchant_id: Option<String>,
     },
+    TotpUpdate {
+        totp_status: Option<TotpStatus>,
+        totp_secret: Option<Encryption>,
+        totp_recovery_codes: Option<serde_json::Value>,
+    },
```

Comment:
If you're expecting to update all 3 fields at once, you can make them all required fields. If there's a need to update only a subset of these fields, you can keep them optional.